app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

Comment:
Email regex is notoriously a difficult thing. See https://emailregex.com/ for an intro.

Reply:
Thank you for this. Right now, I've been referring to Hartl's tutorial for help with Regex. It's something I'm not too familiar with. This is super helpful though. :)

Reply:
Make sure to check out https://rubular.com/ for general ruby regex hacking. I almost always use it when I need to figure one out.

Reply:
Thank you so much!